packages/smooth_app/lib/cache/files/files_cache.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:typed_data';
+
+import 'package:path_provider/path_provider.dart';
+import 'package:smooth_app/cache/cache_manager.dart';
+import 'package:smooth_app/cache/files/file_cache_manager_impl.dart';
+
+/// Files caches work with a [String] key.
+/// This key being the filename.
+typedef FileCache = CacheMap<String, Uint8List>;
+
+/// Cache allowing to store files in the device persistent storage
+class FileCacheManager {
+  const FileCacheManager._();
+
+  static FileCache? _temporaryCache;
+  static FileCache? _longLivingCache;
+
+  static FileCache get temporary {
+    _temporaryCache = FileCacheManagerImpl(
+      rootDirectory: getTemporaryDirectory(),
+      subFolderName: 'temp',
+    );
+
+    return _temporaryCache!;
+  }
+
+  static FileCache get persistent {
+    _longLivingCache = FileCacheManagerImpl(
+      rootDirectory: getApplicationDocumentsDirectory(),
+      subFolderName: 'persistent',
+    );
+
+    return _longLivingCache!;
+  }
+}
```

Comment:
Even if you don't implement it, please add a `TODO` mentioning the possibility of calling an `init` method from `main.dart`.

Reply:
Mmm there is no `init` method.
Which one do you refer to?

Reply:
As we're talking about fields that have an `async` initialization - the physical folder creation at the very first time - it would have made sense to put both folder creations in a `Future init` method and to call it in `main.dart`.

It looks like you decided to init the folders "just in time", which is another point of view.
Let me just say that it does make sense for the "persistent" folder - some users will never scan offline.
But for the "temporary" folder, I see no use-case where a user won't use it, and an explicit `init` method would save time during the very first call.

Anyway, as I said, different points of view, and we're not talking about time-expensive actions (I hope so), so let's keep things the way they are.